crates/oxc_linter/src/rules/eslint/no_octal_escape.rs

```diff
@@ -0,0 +1,172 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+use std::sync::OnceLock;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_octal_escape_diagnostic(span: Span, sequence: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("Don't use octal: '\\{sequence}'. Use '\\u....' instead."))
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoOctalEscape;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallows the use of octal escape sequences in string literals.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Octal escape sequences are deprecated and can lead to unexpected behavior. Modern code should use Unicode or hexadecimal escape sequences instead.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// const str = "\251";
+    /// const str = "\1";
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// const str = "\u00A9";
+    /// const str = "\xA9";
+    /// ```
+    NoOctalEscape,
+    correctness,
```

Comment:
i think should be possible to have a fixer for this?
could we mark it as pending (unless it's not possible to fix?)

Reply:
I think this can be auto-fixed, but we need to be careful because it’s impossible to mechanically determine whether the missing backquote was intentional or if `u.....` should be used instead.

I’ve marked it as pending for now, but the fixer should be implemented as a dangerous auto-fix.

https://github.com/oxc-project/oxc/pull/8151/commits/ac4226fe5559e23cb9effcd95c3a2cea470585e9